OADP-71: Introduction (#38238)
openshift/openshift-docs: backup_and_restore/index.adoc

* You have lost the majority of your control plane hosts, leading to etcd quorum loss.

You can always recover from a disaster situation by xref:../backup_and_restore/control_plane_backup_and_restore/disaster_recovery/scenario-2-restoring-cluster-state.adoc#dr-restoring-cluster-state[restoring your cluster to its previous state] using the saved etcd snapshots.

[id="application-backup-restore-operations-overview"]
== Application backup and restore operations

As a cluster administrator, you can back up and restore applications running on {product-title} by using the OpenShift API for Data Protection (OADP).

> OADP backs up and restores Kubernetes resources, internal images, and cluster metadata, at the granularity of a namespace, by using a supported version of link:https://velero.io/[Velero]. See xref:../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-features_oadp-features-plugins[OADP features].

Reviewer: I think that we may want to consider removing cluster metadata, as I don't know which metadata specifically oadp/velero will backup.

OADP backs up and restores PVs by using the cloud provider snapshot API. You can install OADP with xref:../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[default plugins].

.OADP requirements

OADP has the following requirements:

* You must be logged in as a user with a `cluster-admin` role.
* You must have a supported object storage for storing backups:

** Amazon Web Services
** Microsoft Azure
** Google Cloud Platform
** Multicloud Object Gateway, a component of OpenShift Container Storage
** S3-compatible object storage, for example, Noobaa or Minio

> * You must have a cloud provider that supports snapshots:

Reviewer 2: Unsure if we want this to say "you must". Technically a user can default to restic for filesystem backup/restore.

Author: I was going to change this to "cloud provider that supports snapshots or the Container Snapshot Interface".

Re: Restic. I have not mentioned it because I have not heard that we are documenting it for GA. Can you confirm one way or the other?

[id="application-backup-restore-operations-overview"]
== Application backup and restore operations

As a cluster administrator, you can back up and restore applications running on {product-title} by using the OpenShift API for Data Protection (OADP).

OADP backs up and restores Kubernetes resources and internal images, at the granularity of a namespace, by using link:https://velero.io/[Velero 1.7]. OADP backs up and restores persistent volumes (PVs) by using snapshots or Restic. For details, see xref:../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-features_oadp-features-plugins[OADP features].

You can install OADP with xref:../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[default plugins] that support cloud providers, Container Snapshot Interface (CSI) snapshots, and {product-title} resources.

.OADP requirements

OADP has the following requirements:

* You must be logged in as a user with a `cluster-admin` role.

> * You must have object storage for storing backups:

Reviewer 3: I think the language here should clarify what this list does. Right now, I can't answer these questions:
Do we need ALL the items here? 
Can I use only ONE item?
Can I use another object storage service, or must I use one from this list?

Author: I will add "such as one of the following examples" or similar.

The last item says "S3-compatible", so, yes, you can use any object storage as long as it supports S3. I think that covers all the possibilities.

* You have lost the majority of your control plane hosts, leading to etcd quorum loss.

You can always recover from a disaster situation by xref:../backup_and_restore/control_plane_backup_and_restore/disaster_recovery/scenario-2-restoring-cluster-state.adoc#dr-restoring-cluster-state[restoring your cluster to its previous state] using the saved etcd snapshots.

[id="application-backup-restore-operations-overview"]
== Application backup and restore operations

As a cluster administrator, you can back up and restore applications running on {product-title} by using the OpenShift API for Data Protection (OADP).

> OADP backs up and restores Kubernetes resources and internal images, at the granularity of a namespace, by using a supported version, 1.7, of link:https://velero.io/[Velero]. See xref:../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-features_oadp-features-plugins[OADP features].

Reviewer 3: "at the granularity of a namespace"
I don't understand what this means.

"by using a supported version, 1.7, of link:https://velero.io/[Velero]"
This is a bit confusing. Seems to me that "supported version" and "1.7" don't work together. Can we just say "Valero"? 

"See xref:../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-features_oadp-features-plugins[OADP features]."
It's not clear that the preceding sentence was describing "OADP features", so the link text makes me pause. Perhaps something along the lines of "For more details of what OADP does, see link:blablabla [OADP features]."

Author: > "at the granularity of a namespace" I don't understand what this means.

Backup and restore can be used for cluster resources. Here, we want the user to know that they can back up a single namespace.

> "by using a supported version, 1.7, of link:[https://velero.io/[Velero]](https://velero.io/%5BVelero%5D)" This is a bit confusing. Seems to me that "supported version" and "1.7" don't work together. Can we just say "Valero"?

Not all versions of Velero are supported. Velero is an upstream open-source project that we have been using for MTC. That is why we have to specify that it is a supported version. I can rewrite this sentence to make it smoother.

> "See xref:../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-features_oadp-features-plugins[OADP features]." It's not clear that the preceding sentence was describing "OADP features", so the link text makes me pause. Perhaps something along the lines of "For more details of what OADP does, see link:blablabla [OADP features]."

The previous section was basic functionality. Features includes other functionality that the user might not use, for example, hooks and scheduling. Will check to see whether I can make it clearer.

> OADP backs up and restores PVs by using the cloud provider snapshot API, Container Snapshot Interface (CSI) snapshots, or link:https://restic.net/[Restic].

Reviewer 3: Does "PV" need to be spelled out on first ref?

You can install OADP with xref:../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[default plugins].

> .OADP requirements

Reviewer 3: I'm a bit confused by this section, probably because I normally expect this kind of information before a procedure. Other OCP components list these sorts of requirements before you _do_ something. 

So I'm left wondering. What are these requirements for exactly?

Author: This is the landing page for the entire Application Backup and Restore section of the guide. We have to tell them up front what they need. Otherwise, they will waste time going to the procedure to see that they need S3 object storage or snapshot-enabled cloud storage.

Author: Also, the user needs to know that they need to be cluster admins.

OADP has the following requirements:

* You must be logged in as a user with a `cluster-admin` role.
* You must have object storage for storing backups:

** Amazon Web Services
** Microsoft Azure
** Google Cloud Platform
** Multicloud Object Gateway, a component of OpenShift Container Storage
** S3-compatible object storage, for example, Noobaa or Minio

:FeatureName: The `CloudStorage` API for S3 storage
include::modules/technology-preview.adoc[]

> * You must have a cloud provider that has a native snapshot API or supports CSI snapshots if you want to back up PVs with snapshots:

Reviewer 3: This lead-in sentence could be stronger if were more specific. It hints at what's coming, but doesn't say it explicitly. I don't know if the list provides examples or a complete list of providers. 
Suggestion:
*  If you want to back up PVs with snapshots, you must have a cloud provider that has a native snapshot API or supports CSI snapshots, such as the following providers:

** Amazon Web Services
** Microsoft Azure
** Google Cloud Platform

> ** CSI-enabled storage, for example, Ceph RBD or Ceph FS

Reviewer 3: ```suggestion
** CSI-enabled storage, such as Ceph RBD or Ceph FS
```
On first read, I saw " CSI-enabled storage, for example."

Reviewer 3: To fit in the list, perhaps this should be reworded to something like this:

* any provider that uses CSI-enabled storage, such as Ceph RBD or Ceph FS

Author: OK. I can do that. I also saw that I forgot to mention that it's not just CSI but CSI snapshots, so I will fix that.
